flood_api/routers/flood.py

```diff
@@ -1,21 +1,125 @@
-from fastapi import APIRouter
+from fastapi import APIRouter, HTTPException
+from fastapi.responses import JSONResponse
 
-from flood_api.dependencies.flooddata import DetailedDataDep
-from flood_api.dependencies.queryparams import CoordinatesDep, DateRangeDep
+from flood_api.dependencies.flooddata import (DetailedDataDep,
+                                              SummaryDataDep,
+                                              ThresholdDataDep)
+from flood_api.dependencies.queryparams import (CoordinatesDep, 
+                                                DateRangeDep,
+                                                IncludeNeighborsDep)
+from flood_api.utils.geospatial_operations import get_data_for_point
+from flood_api.utils.json_utilities import (dataframe_to_geojson, 
+                                            custom_date_handler)
+from flood_api.utils.validation_helpers import point_within_roi
 
 router = APIRouter(tags=["flood"])
 
+@router.get("/summary", response_class=JSONResponse)
+async def summary(
+    gdf: SummaryDataDep,
+    coordinates: CoordinatesDep,
+    include_neighbors: IncludeNeighborsDep,
+):
+    longitude, latitude = coordinates
+
+    # Check if the queried point is within the ROI
+    if not point_within_roi(latitude=latitude, longitude=longitude):
+        raise HTTPException(status_code=400, detail="Queried coordinates are outside the region of interest")
+
+    queried_cell, neighboring_cells = get_data_for_point(latitude=latitude, longitude=longitude, 
+                                                         include_neighbors=include_neighbors, gdf=gdf)
+
+    summary_cols = [
```

Comment:
All these columns that are returned should be part of a pydantic model and be thoroughly documented.